snuba/admin/static/components/query_display/types.tsx

```diff
@@ -24,6 +24,8 @@ type QueryResult = {
   timestamp: number;
   column_names: QueryResultColumnMetadata;
   rows: [QueryResultRow];
+  trace_output?: string;
+  error?: string;
```

Comment:
I would not try and make every kind of API response fit into this QueryResult type by making lots of stuff optional.
I think it is better to have 3 separate and stricter types.
- SystemQueryResult
- TracingQueryResult (this one has the trace_output)
- QueryResultError (this one only has the error key)

Reply:
I don't think I agree to this. If we make different return types for different functions, then the common QueryDisplay component has to also know which function is being called and the type it returns, which means that has to be passed in the props and is going to make the nice common component Rahul created a lot messier. 

Also these types are not independent, a system query result could have a tracing output and/or an error. This type maps to the response from Snuba, which can have any combination of these fields. Saying that one call to Snuba will necessarily only have a subset of fields isn't strictly true.